lib/git-fastclone.rb

```diff
@@ -333,24 +338,26 @@ def update_reference_repo(url, fail_hard)
         submodule_file = reference_repo_submodule_file(url, reference_dir, using_local_repo)
 
         # if prefetch is on, then grab children immediately to frontload network requests
-        prefetch(submodule_file) if File.exist?(submodule_file) && prefetch_submodules
+        prefetch(submodule_file, attempt_number) if File.exist?(submodule_file) && prefetch_submodules
 
         # Store the fact that our repo has been updated if necessary
-        store_updated_repo(url, mirror, repo_name, fail_hard) unless reference_updated[repo_name]
+        store_updated_repo(url, mirror, repo_name, fail_hard, attempt_number) unless reference_updated[repo_name]
       end
     end
 
     # Grab the children in the event of a prefetch
-    def prefetch(submodule_file)
+    def prefetch(submodule_file, attempt_number)
       File.readlines(submodule_file).each do |line|
         # We don't join these threads explicitly
-        Thread.new { update_reference_repo(line.strip, false) }
+        Thread.new { update_reference_repo(line.strip, false, attempt_number) }
       end
     end
 
     # Creates or updates the mirror repo then stores an indication
     # that this repo has been updated on this run of fastclone
-    def store_updated_repo(url, mirror, repo_name, fail_hard)
+    def store_updated_repo(url, mirror, repo_name, fail_hard, attempt_number)
+      trigger_pre_clone_hook(url, mirror, attempt_number)
```

Comment:
We might want to disambiguate which `clone` operation we're referring to. e.g. `trigger_pre_mirror_clone_hook`? Should we only call this when we think the clone might actually be called next (e.g. within the `unless` block)?

Reply:
If you think that would be better, let me do that.